### PR TITLE
feat: separate api call from react component

### LINKED ISSUE #2812: [Refactor] API calls | developers should not make direct API calls in React components / pages
The issue comes from this PR #2715 

Screenshot from 2024-07-30 23-36-29

### PR DESCRIPTION
**Pull request by CREDO23**
### This PR fixes #2812 

[Loom](https://www.loom.com/share/ac6c731d4a274c44a494c66dd6c557b4?sid=400ef03a-b9ef-4858-bcde-c532ef3dba9d)

### CODE REVIEW

File: apps/web/lib/features/manual-time/add-manual-time-modal.tsx [hunk 1]
```diff
@@ -224,10 +225,10 @@ export function AddManualTimeModal(props: IAddManualTimeModalProps) {
 						Team<span className="text-[#de5505e1] ml-1">*</span>
 					</label>
 					<SelectItems
+						defaultValue={activeTeam!}
 						items={teams}
-						defaultValue={activeTeam}
-						onValueChange={(value) => setTeamId(value ? value.id : '')}
-						itemId={(team) => (team ? team.id : '')}
+						onValueChange={(team) => setTeam(team)}
+						itemId={(team) => (team ? team.name : '')}
```
Comment (evereq) [inline apps/web/lib/features/manual-time/add-manual-time-modal.tsx:231]: @CREDO23 why the team name become Id!??

File: apps/web/lib/features/manual-time/add-manual-time-modal.tsx [hunk 2]
```diff
@@ -47,40 +51,30 @@ export function AddManualTimeModal(props: IAddManualTimeModalProps) {
...
 			e.preventDefault();
 
-			const timeObject = {
-				date,
-				isBillable,
-				startTime,
-				endTime,
-				teamId,
+			const startedAt = new Date(date);
+			const stoppedAt = new Date(date);
```
Comment (evereq) [inline apps/web/lib/features/manual-time/add-manual-time-modal.tsx:55]: @CREDO23 I am not getting logic here... can you explain? Why date is same and only hours / minutes we set!? What is this for???

File: apps/web/lib/features/manual-time/add-manual-time-modal.tsx [hunk 3]
```diff
@@ -47,40 +51,30 @@ export function AddManualTimeModal(props: IAddManualTimeModalProps) {
...
+				stoppedAt,
 				taskId,
 				description,
 				reason,
-				timeDifference
+				isBillable,
+				organizationId: team?.organizationId
 			};
 
-			if (date && startTime && endTime && teamId && taskId) {
```
Comment (evereq) [inline apps/web/lib/features/manual-time/add-manual-time-modal.tsx:62]: We had condition logic @CREDO23 , "if (date && startTime && endTime && teamId && taskId) {" why you removed it!?

File: apps/web/lib/features/manual-time/add-manual-time-modal.tsx [hunk 4]
```diff
@@ -47,40 +51,30 @@ export function AddManualTimeModal(props: IAddManualTimeModalProps) {
...
-						}
-					} catch (err) {
-						setError('Failed to post data');
-						setLoading(false);
-					}
-				};
-
-				postData();
-			} else {
-				setError(`Please complete all required fields with a ${'*'}`);
```
Comment (evereq) [inline apps/web/lib/features/manual-time/add-manual-time-modal.tsx:80]: @CREDO23 it was some validation message here? How it's handled now!?

File: apps/web/lib/features/manual-time/add-manual-time-modal.tsx (hunk 2 above)
Comment (CREDO23) [reply apps/web/lib/features/manual-time/add-manual-time-modal.tsx:55]: The date will be the same @evereq :
Screenshot from 2024-08-02 20-22-46

File: apps/web/lib/features/manual-time/add-manual-time-modal.tsx (hunk 1 above)
Comment (CREDO23) [reply apps/web/lib/features/manual-time/add-manual-time-modal.tsx]: It's fixed @evereq

File: apps/web/lib/features/manual-time/add-manual-time-modal.tsx (hunk 3 above)
Comment (CREDO23) [reply apps/web/lib/features/manual-time/add-manual-time-modal.tsx:62]: I just added it! @evereq

File: apps/web/lib/features/manual-time/add-manual-time-modal.tsx (hunk 4 above)
Comment (CREDO23) [reply apps/web/lib/features/manual-time/add-manual-time-modal.tsx:80]: I just added it @evereq

File: apps/web/lib/features/manual-time/add-manual-time-modal.tsx (hunk 2 above)
Comment (Cedric921) [reply apps/web/lib/features/manual-time/add-manual-time-modal.tsx:55]: Initially and When `Start Time` change, `End Time` must be must be 10 minutes longer than `Start Time`

**Pull request merged**